rest-api/authentication/models/auth-models.js

```diff
@@ -0,0 +1,35 @@
+const db = require('../../../database/dbConfig');
+const bcrypt = require('bcryptjs');
+const { generateToken } = require('../middlewares/generateToken');
+
+function register(user) {
+    return db('users')
+        .insert(user)
+        .returning('*')
+        .then(([user]) => {
+            delete user.password;
```

Comment:
I think the delete functionality is okay, if it works as it's supposed to. Otherwise, it would be best to simply return the just the needed fields instead of (*)